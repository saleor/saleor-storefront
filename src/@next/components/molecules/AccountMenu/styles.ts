import { styled } from "@styles";

export const Wrapper = styled.div`
  background-color: ${props => props.theme.colors.light};
  height: 100%;
  /* height: auto; */
  padding-left: 3rem;
  padding-top: 2.5rem;
`;

export const MenuHeader = styled.div`
  font-size: ${props => props.theme.typography.h3FontSize};
  font-weight: ${props => props.theme.typography.boldFontWeight};
  text-transform: "uppercase";
  padding-bottom: 2rem;
`;

export const MenuItem = styled.div<{
  active: boolean;
}>`
  cursor: pointer;
  padding-bottom: 1.5rem;
  color: ${props => (props.active ? props.theme.colors.activeMenuOption : "")};
`;
