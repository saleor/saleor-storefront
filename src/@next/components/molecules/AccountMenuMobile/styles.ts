import { styled } from "@styles";

export const Wrapper = styled.div`
  background-color: ${props => props.theme.colors.light};
  padding: 1.25rem;
  font-weight: ${props => props.theme.typography.boldFontWeight};
  display: flex;
  justify-content: space-between;
  align-items: center;
  position: relative;
  margin-bottom: 2rem;
`;

export const Overlay = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  padding: 1.25rem;
  height: auto;
  /* height: 350px; */
  overflow: visible;
  z-index: 10;
  background-color: white;
  box-shadow: 0 0 0 9999px rgba(50, 50, 50, 0.1);
`;

export const MenuHeader = styled.div`
  font-weight: ${props => props.theme.typography.boldFontWeight};
  padding-bottom: 2rem;
`;

export const MenuItem = styled.div<{
  active: boolean;
}>`
  display: flex;
  justify-content: space-between;
  padding-bottom: 1.5rem;
  color: ${props => (props.active ? props.theme.colors.activeMenuOption : "")};

  svg {
    transform: rotate(-90deg);
  }
`;
