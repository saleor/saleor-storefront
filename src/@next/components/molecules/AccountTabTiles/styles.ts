import { media, styled } from "@styles";

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  align-items: center;
  justify-content: flex-start;
`;

export const TileWrapper = styled.div`
  height: auto;
  margin-bottom: 1.5rem;
`;

export const Header = styled.div`
  width: 95%;
  padding-bottom: 1rem;
  border-bottom: 1px solid ${props => props.theme.colors.dividerDark};
  font-weight: ${props => props.theme.typography.boldFontWeight};
  font-size: ${props => props.theme.typography.h4FontSize};
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 4rem;
`;

export const HeaderSmall = styled(Header)`
  width: 100%;
  border-bottom: none;
`;

export const Content = styled.div`
  padding: 1.5rem 0;
  width: 95%;
`;

export const ContentOneLine = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 70%;
  ${media.smallScreen`
    flex-direction: column;
    width: 100%;
  `}
`;
export const ContentEdit = styled.div`
  width: 50%;
  ${media.smallScreen`
     width: 100%;
  `}
`;

export const ContentEditOneLine = styled.div`
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  > div {
    width: 48%;
    ${media.smallScreen`
      width: 100%;
    `}
  }

  ${media.smallScreen`
     flex-direction: column;
  `}
`;

export const ContentExtendInput = styled.div`
  width: 60%;
`;

export const Form = styled.form`
  background-color: ${props => props.theme.tile.backgroundColor};
`;

export const FormButtons = styled.div`
  height: 5rem;
  padding-top: 2rem;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  button {
    margin-left: 2rem;
  }
`;
