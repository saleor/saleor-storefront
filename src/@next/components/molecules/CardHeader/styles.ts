import { styled } from "@styles";

import { TitleSize } from "./types";

export const Wrapper = styled.div<{ divider: boolean }>`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: ${props => `1.1rem ${props.theme.spacing.gutter}`};
  ${({ divider, theme }) =>
    divider && `border-bottom: 1px solid ${theme.colors.light};`}
`;

export const Header = styled.div`
  display: flex;
  align-items: center;
  gap: ${props => props.theme.spacing.gutter};
`;

export const Title = styled.h4<{ size: TitleSize }>`
  font-size: ${({ size, theme: { typography } }) =>
    size === "lg" ? typography.h4FontSize : typography.baseFontSize};
  font-weight: ${props => props.theme.typography.boldFontWeight};
  color: ${props => props.theme.colors.baseFont};
  text-transform: uppercase;
  padding-right: 0.6rem;
  margin: 0;
`;

export const Paragraph = styled.p`
  font-size: ${props => props.theme.typography.smallFontSize};
  color: ${props => props.theme.colors.lightFont};
  padding-right: 0.6rem;
  margin: 0;
`;
