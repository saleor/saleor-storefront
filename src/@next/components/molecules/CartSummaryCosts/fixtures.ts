export const DEFAULT_PROPS = {
  subtotalPrice: "150$",
  totalPrice: "180$",
};
