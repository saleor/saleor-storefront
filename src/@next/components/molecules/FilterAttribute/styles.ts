import { styled } from "@styles";

export const Wrapper = styled.div`
  width: 80%;
  padding-bottom: 2rem;
`;

export const Header = styled.div`
  font-size: ${props => props.theme.typography.h4FontSize};
  font-weight: ${props => props.theme.typography.boldFontWeight};
  padding-bottom: 1.5rem;
`;

export const Checkbox = styled.div`
  width: 100%;
  margin-bottom: 1.25rem;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  cursor: pointer;
  position: relative;
  margin-left: -4px;
`;

export const BottomBorder = styled.div`
  border-bottom: 1px solid ${props => props.theme.colors.divider};
  width: 95%;
`;

export const ViewMoreButton = styled.div`
  padding-bottom: 1.25rem;
`;

export const Label = styled.label`
  display: flex;
  cursor: pointer;
  justify-content: flex-start;
  align-items: center;
  padding-right: 1.25rem;
  input[type="checkbox"] {
    display: none;
    position: relative;
    right: -999em;
  }
  div {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 26px;
    width: 26px;

    span {
      border: 1px solid ${props => props.theme.colors.secondary};
      width: 14px;
      height: 14px;
      display: inline-block;
      z-index: 10;
    }

    ${Checkbox}:hover & {
      border-radius: 50%;
      border: 1px solid;
      border-color: ${props => props.theme.colors.secondaryOverlay};
      background-color: ${props => props.theme.colors.secondaryOverlay};
    }

    :focus {
      border-radius: 50%;
      border: 1px solid;
      outline: none;
      border-color: ${props => props.theme.colors.secondaryOverlayDark};
      background-color: ${props => props.theme.colors.secondaryOverlayDark};
    }
  }

  input:checked + div {
    span {
      background-clip: content-box;
      padding: 2px;
      background-color: ${props => props.theme.colors.secondary};
    }
  }
`;
