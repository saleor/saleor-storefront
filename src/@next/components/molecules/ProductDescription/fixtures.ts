export const attributes = [
  {
    attribute: { name: "Author" , translation: {
      __typename: "AttributeTranslation",
      name: "t",
    },
  },
  values: [
    {
      name: "John Doe",
      translation: {
        __typename: "AttributeValueTranslation",
        name: "t",
      },
    },
    
  ],
},
{
  attribute: { name: "Country of Manufacture" , translation: {
    __typename: "AttributeTranslation",
    name: "t",
  },
},
values: [
  {
    name: "Poland",
    translation: {
      __typename: "AttributeValueTranslation",
      name: "t",
    },
  },
  
  {
    name: "China",
    translation: {
      __typename: "AttributeValueTranslation",
      name: "t",
    },
  },
  
],
},
{
  attribute: { name: "Size" , translation: {
    __typename: "AttributeTranslation",
    name: "t",
  },
},
values: [
  {
    name: "40x40cm",
    translation: {
      __typename: "AttributeValueTranslation",
      name: "t",
    },
  },
  
],
},
{
  attribute: { name: "Material" , translation: {
    __typename: "AttributeTranslation",
    name: "t",
  },
},
values: [
  {
    name: "Polystyrene",
    translation: {
      __typename: "AttributeValueTranslation",
      name: "t",
    },
  },
  
],
},
{
  attribute: { name: "Manufacture time" , translation: {
    __typename: "AttributeTranslation",
    name: "t",
  },
},
values: [
  {
    name: "2 to 4 weeks",
    translation: {
      __typename: "AttributeValueTranslation",
      name: "t",
    },
  },
  
],
},
{
  attribute: { name: "SKU" , translation: {
    __typename: "AttributeTranslation",
    name: "t",
  },
},
values: [
  {
    name: "1244-SGG",
    translation: {
      __typename: "AttributeValueTranslation",
      name: "t",
    },
  },
  
],
},
];

export const description =
"This is some cool product description. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.";
