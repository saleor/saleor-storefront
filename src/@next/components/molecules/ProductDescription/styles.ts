import { media, styled } from "@styles";
import { orange, white } from "@styles/constants";

export const Wrapper = styled.div``;

export const AttributeList = styled.ul`
  columns: 2;
  column-width: 50%;

  ${media.largeScreen`
    column-width: 100%;
    columns: 1;
  `};
  width: 100%;
  padding: 0;
  margin: 0;

  li {
    margin-bottom: 30px;
    font-size: ${props => props.theme.typography.h4FontSize};
  }

  li::before {
    content: "â€¢";
    margin-right: 20px;
    color: ${props => props.theme.colors.listBullet};
  }
`;

export const Tabs = styled.div`
  display: flex;
  flex-wrap: none;
  width: 100%;
  overflow: hidden;
`;

export const TabTitle = styled.div<{ active?: boolean }>`
  cursor: pointer;
  font-size: ${props => props.theme.typography.baseFontSize};
  letter-spacing: 0.02em;
  color: ${props => props.active && props.theme.colors.tabTitle};
  border-bottom-width: ${props => (props.active ? "4px" : "0")};
  border-bottom-style: solid;
  border-bottom-color: ${props =>
    props.active ? props.theme.colors.tabTitle : "transparent"};
  padding: 12px 16px;
  margin-right: 8px;
  border-top: 1px solid #dcdee3;
  border-left: 1px solid #dcdee3;
  border-right: 1px solid #dcdee3;
  border-radius: 8px 8px 0 0;
  background-color: ${props => (props.active ? "#ffffff" : "transparent")};

  ${media.smallScreen`
    font-size: ${(props: any) => props.theme.typography.baseFontSize};
  `};
`;

export const AttributeName = styled.span`
  color: ${props => props.theme.colors.listAttributeName};
`;

export const WrapBox = styled.div`
  padding: 0 1rem;
`;
export const CompanyImage = styled.div`
  display: flex;
  flex-direction: column;
`;
export const TitleText = styled.h3`
  font-size: 16px;
  font-weight: 700;
  margin-bottom: 16px;
  text-transform: uppercase;
`;
// export const SubTitleText = styled.h4`
//   font-size: 14px;
//   margin-bottom: 20px;
//   color: #666;
// `;
export const CompanyWrapInfo = styled.div`
  position: relative;
`;
export const CompanyInfo = styled.table`
  width: 100%;
  margin: 1rem 0;
`;
export const CompanyInfoBody = styled.tbody``;
export const CompanyInfoTr = styled.tr`
  border-bottom: none;
`;
export const CompanyInfoTd = styled.td`
  background-color: #fafafa;
  border: 1px solid #f4f4f4;
  &: nth-child(2n) {
    background-color: #fff;
  }
`;
// export const CompanyLocation = styled.div`
//   display: flex;
//   flex-direction: column;
// `;
export const WrapperContent = styled.div`
  background-color: #ffffff;
  border: 1px solid #cccccc;
  padding: 12px;
`;

export const ButtonStore = styled.div`
  padding: 0.75rem;
  border-radius: 2rem;
  background: ${orange};
  color: ${white};
  cursor: pointer;
`;
