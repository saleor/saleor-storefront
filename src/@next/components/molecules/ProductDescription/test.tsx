import "jest-styled-components";

import { mount, shallow } from "enzyme";
import React from "react";

import { ProductDescription } from ".";
import { attributes, description } from "./fixtures";
import * as S from "./styles";

describe("<ProductDescription />", () => {
  it("exists", () => {
    const wrapper = shallow(
      <ProductDescription attributes={attributes} description={description} />
    );

    expect(wrapper.exists()).toEqual(true);
  });

  it("should contain and show by default product description", () => {
    const wrapper = shallow(
      <ProductDescription attributes={attributes} description={description} />
    );

    expect(wrapper.text()).toContain(description);
  });

  it("should show product attributes when clicking on attributes tab", () => {
    const wrapper = mount(
      <ProductDescription attributes={attributes} description={description} />
    );

    wrapper.find(S.TabTitle).at(1).simulate("click");

    attributes.forEach(attribute =>
      expect(wrapper.text()).toContain(attribute.attribute.name)
    );
  });
});
