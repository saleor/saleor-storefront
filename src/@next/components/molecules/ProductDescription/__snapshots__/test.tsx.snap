// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDescription /> renders correctly 1`] = `
.c3 .ce-block__content {
  max-width: 100%;
}

.c4 {
  -webkit-columns: 2;
  columns: 2;
  -webkit-column-width: 50%;
  column-width: 50%;
  width: 100%;
  padding: 0;
  margin: 0;
}

.c4 li {
  margin-bottom: 30px;
  font-size: 1.125rem;
}

.c4 li::before {
  content: "â€¢";
  margin-right: 20px;
  color: #06847B;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: none;
  -ms-flex-wrap: none;
  flex-wrap: none;
  width: 100%;
  border-bottom: 1px solid rgba(50,50,50,0.1);
  margin-bottom: 70px;
  overflow: hidden;
}

.c1 {
  cursor: pointer;
  min-width: 230px;
  font-size: 1.5rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.02em;
  -moz-letter-spacing: 0.02em;
  -ms-letter-spacing: 0.02em;
  letter-spacing: 0.02em;
  color: #06847B;
  border-bottom-width: 4px;
  border-bottom-style: solid;
  border-bottom-color: #06847B;
  padding-bottom: 25px;
  margin-right: 60px;
}

.c2 {
  cursor: pointer;
  min-width: 230px;
  font-size: 1.5rem;
  font-weight: 600;
  -webkit-letter-spacing: 0.02em;
  -moz-letter-spacing: 0.02em;
  -ms-letter-spacing: 0.02em;
  letter-spacing: 0.02em;
  border-bottom-width: 4px;
  border-bottom-style: solid;
  border-bottom-color: transparent;
  padding-bottom: 25px;
  margin-right: 60px;
}

.c5 {
  color: rgba(50,50,50,0.6);
}

@media (max-width:992px) {
  .c4 {
    -webkit-column-width: 100%;
    column-width: 100%;
    -webkit-columns: 1;
    columns: 1;
  }
}

@media (max-width:540px) {
  .c1 {
    font-size: 1.125rem;
    min-width: 150px;
    margin-right: 20px;
  }
}

@media (max-width:540px) {
  .c2 {
    font-size: 1.125rem;
    min-width: 150px;
    margin-right: 20px;
  }
}

<div
  className=""
>
  <div
    className="c0"
  >
    <div
      className="c1"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      DESCRIPTION
    </div>
    <div
      className="c2"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      ATTRIBUTES
    </div>
  </div>
  <div
    hidden={false}
  >
    <div
      className="c3"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Tonight, my love, let us take fat brushes and paint the skies with the shades of nebula tides. Get the dark, moody shades of the ether.</p>",
        }
      }
    />
  </div>
  <div
    hidden={true}
  >
    <ul
      className="c4"
    >
      <li>
        <span
          className="c5"
        >
          Author
          : 
        </span>
         
        John Doe
      </li>
      <li>
        <span
          className="c5"
        >
          Country of Manufacture
          : 
        </span>
         
        Poland, China
      </li>
      <li>
        <span
          className="c5"
        >
          Size
          : 
        </span>
         
        40x40cm
      </li>
      <li>
        <span
          className="c5"
        >
          Material
          : 
        </span>
         
        Polystyrene
      </li>
      <li>
        <span
          className="c5"
        >
          Manufacture time
          : 
        </span>
         
        2 to 4 weeks
      </li>
      <li>
        <span
          className="c5"
        >
          SKU
          : 
        </span>
         
        1244-SGG
      </li>
    </ul>
  </div>
</div>
`;
